src/datasources/asset/components/editors/asset-summary/AssetSummaryDataSource.ts

```diff
@@ -17,10 +19,35 @@ export class AssetSummaryDataSource extends AssetDataSourceBase {
     defaultQuery = {
     };
 
-    runQuery(query: AssetQuery, options: DataQueryRequest): Promise<DataFrameDTO> {
-        throw new Error("Method not implemented.");
+    async runQuery(query: AssetQuery, options: DataQueryRequest): Promise<DataFrameDTO> {
+        return this.processMetadataQuery(query as AssetQuery);
+    }
+
+    shouldRunQuery(_: AssetQuery): boolean {
+        return true;
     }
-    shouldRunQuery(query: AssetQuery): boolean {
-        throw new Error("Method not implemented.");
+
+    async processMetadataQuery(query: AssetQuery) {
+        const result: DataFrameDTO = { refId: query.refId, fields: [] };
+
+        const assets: AssetSummaryQuery = await this.getAssetSummary();
+
+        result.fields = [
+            { name: 'Total', values: [assets.total] },
+            { name: 'Active', values: [assets.active] },
+            { name: 'Not active', values: [assets.notActive] },
+            { name: 'Approaching due date', values: [assets.approachingRecommendedDueDate] },
+            { name: 'Past due date', values: [assets.pastRecommendedDueDate] }
+        ];
+        return result;
+    }
+
+    async getAssetSummary(): Promise<AssetSummaryQuery> {
+        try {
+            let response = await this.get<AssetSummaryQuery>(this.baseUrl + '/asset-summary');
+            return response;
+        } catch (error) {
+            throw new Error(`An error occurred while getting asset summary: ${error}`);
```

Comment:
no need to rethrow

Reply:
Isn't this how we handled the main requests in the other data sources as well? Should we change that approach?

Reply:
@kkerezsi  I modified the request for my PR, but just wanted to let you know that we have this approach everywhere. Should we create a tech debt to address this?

Reply:
Let's leave it as is for now and we'll treat it as part of a generic code since it will anyways not display the right thing. 

The error message will be 'An error occurred while getting asset summary: [Object object]'

I've created a task for this